contracts/contracts/strategies/CompoundStrategy.sol

```diff
@@ -194,17 +196,17 @@ contract CompoundStrategy is BaseCompoundStrategy {
     }
 
     /**
-     * @dev Approve the spending of all assets by their corresponding cToken,
+     * @notice Approve the spending of all assets by their corresponding cToken,
      *      if for some reason is it necessary.
      */
     function safeApproveAllTokens() external override {
         uint256 assetCount = assetsMapped.length;
-        for (uint256 i = 0; i < assetCount; i++) {
-            address asset = assetsMapped[i];
-            address cToken = assetToPToken[asset];
+        for (uint256 i = 0; i < assetCount; ++i) {
+            IERC20 asset = IERC20(assetsMapped[i]);
+            address cToken = assetToPToken[address(asset)];
             // Safe approval
-            IERC20(asset).safeApprove(cToken, 0);
-            IERC20(asset).safeApprove(cToken, type(uint256).max);
+            asset.safeApprove(cToken, 0);
```

Comment:
Yes, because of USDT we have to do the approve to zero then approve back up here.

However, we can just call approve on the the token, and remove the use of the depreciated safeApprove from the OZ library.